golem_messages/message.py

```diff
@@ -34,6 +34,44 @@ def verify_time(timestamp):
         raise exceptions.MessageFromFutureError()
 
 
+class FrozenDict(dict):

```

Comment:
https://pypi.python.org/pypi/frozendict ?

Reply:
Same name different functionality :)

Reply:
How about [\_\_slots\_\_](https://stackoverflow.com/questions/472000/usage-of-slots) with default values?

Here's an example (from https://mail.python.org/pipermail/python-dev/2003-May/035607.html):

```python
>>> class Pane(object):
	__slots__ = { 'background': 'black', 'foreground': 'white',
		      'size': (80, 25) }
	def __init__(self):
		for k, v in self.__slots__.iteritems():
			setattr(self, k, v)

			
>>> p = Pane()
>>> p.background = 'blue'
>>> p.background, p.foreground
('blue', 'white')
```

Reply:
It's a nice hack of `__slots__` but it doesn't have the advantage of dictcless instance (still a dict for default values). And it's a hack... I prefer for dicts to be straightforward and look like dicts.

Reply:
So lets do it this way:

- Add some documentation to express what this FrozenDict is about.
- move it to a separate file.
- change `__missing__` to 

```python
from copy import deepcopy

def __missing__(self, key):
    return deepcopy(self.ITEMS[key])
```
This will allow to put dict or anything else as a default value.

Reply:
So lets do it this way:

- Add some documentation to express what this FrozenDict is about.
- move it to a separate file.
- change `__missing__` to 

```python
from copy import deepcopy

def __missing__(self, key):
    return deepcopy(self.ITEMS[key])
```
This will allow to put dict or anything else as a default value.